src/jobsautoreport/models.py

```diff
@@ -8,6 +8,6 @@ class JobType(Enum):
     BATCH = "batch"
 
 
-class JobState(Enum):
+class ObjectState(Enum):
```

Comment:
```suggestion
class JobState(Enum):
...

StepState = NewType("StepState", JobState)
```
The name `ObjectState` is too vague to me. You can create an alias with `NewType` https://docs.python.org/3/library/typing.html#newtype.

Or you can just name it like `ProwExecutionState` maybe?